lib/ansiblelint/utils.py

```diff
@@ -599,55 +599,109 @@ def get_first_cmd_arg(task):
 
 
 def append_skipped_rules(pyyaml_data, file_text, file_type):
-    """ Uses ruamel.yaml to parse comments then adds a
-        skipped_rules list to the task (or meta yaml block)
+    """Append 'skipped_rules' to individual tasks or single metadata block.
+
+    For a file, uses 2nd parser (ruamel.yaml) to pull comments out of
+    yaml subsets, check for '# noqa' skipped rules, and append any skips to the
+    original parser (pyyaml) data relied on by remainder of ansible-lint.
+
+    :param pyyaml_data: file text parsed via ansible and pyyaml.
+    :param file_text: raw file text.
+    :param file_type: type of file: tasks, handlers or meta.
+    :returns: original pyyaml_data altered with a 'skipped_rules' list added
+    to individual tasks, or added to the single metadata block.
     """
 
-    PLAYBOOK_TASK_KEYWORDS = [
-        'tasks',
-        'pre_tasks',
-        'post_tasks',
-        'handlers',
-    ]
+    try:
+        yaml_skip = _append_skipped_rules(pyyaml_data, file_text, file_type)
+    except Exception as exc:
```

Comment:
Could you please specify exact errors which may happen? Wholesale excepts is a bad practice.

Reply:
Agreed generic except should be avoided most times... however here I was thinking that regardless of any possible unexpected error within skipping portion, there is value in continuing program execution

Reply:
The danger of the wholesale exception swallowing is that you'll miss some important errors which may need to be processed in a better way. It also breaks the debugging experience.
+ Nobody is going to report such errors back.

Reply:
Fair point, I made the error specific